File: web/src/update_drag.py
import os
import glob
import re

files = glob.glob('c:/Users/Win10/Desktop/reactnative-etala/web/src/pages/superadmin/*.jsx')

count = 0
for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # We are looking for something like:
    # <input
    #  type="file"
    #  ...
    #  className="hidden"
    # />
    
    # Regex to find an input tag with className="hidden" and type="file"
    # Since React code varies, we can find className="hidden" inside an input tag.
    
    # Actually, a simpler replace:
    # 1. find `<input...className="hidden"...type="file"`
    # 2. Or just replace `className="hidden"` with `className="absolute inset-0 w-full h-full opacity-0 cursor-pointer z-50"`
    # BUT only for the inputs that are file uploads.
    
    new_content = content
    # Find all <input ... />
    inputs = re.findall(r'<input[^>]+type=["\']file["\'][^>]*className=["\']hidden["\'][^>]*>', content)
    
    modified = False
    for inp in inputs:
        new_inp = inp.replace('className="hidden"', 'className="absolute inset-0 w-full h-full opacity-0 cursor-pointer z-50"')
        new_content = new_content.replace(inp, new_inp)
        modified = True
        
    inputs2 = re.findall(r'<input[^>]*className=["\']hidden["\'][^>]*type=["\']file["\'][^>]*>', content)
    for inp in inputs2:
        new_inp = inp.replace('className="hidden"', 'className="absolute inset-0 w-full h-full opacity-0 cursor-pointer z-50"')
        new_content = new_content.replace(inp, new_inp)
        modified = True

    # We also need to add 'relative' to the container of this input.
    # The container is usually a <div className="border-2 border-dashed... "> that immediately precedes or wraps this.
    # We can replace `className="border-2 border-dashed` with `className="relative border-2 border-dashed`
    if modified:
        new_content = new_content.replace('className="border-2 border-dashed', 'className="relative border-2 border-dashed')
        
        with open(file, 'w', encoding='utf-8') as f:
             f.write(new_content)
        count += 1
        print("Updated", file)

print(f"Total updated: {count}")

File: web/src/update_drag.js
const fs = require('fs');
const glob = require('glob');
const parser = require('@babel/parser');
const traverse = require('@babel/traverse').default;
const generate = require('@babel/generator').default;

const files = glob.sync('c:/Users/Win10/Desktop/reactnative-etala/web/src/pages/**/*.{jsx,js}');

let updatedCount = 0;

for (const file of files) {
    let code = fs.readFileSync(file, 'utf8');
    let ast;

    try {
        ast = parser.parse(code, {
            sourceType: 'module',
            plugins: ['jsx']
        });
    } catch (err) {
        console.error(`Error parsing ${file}:`, err);
        continue;
    }

    let modified = false;

    traverse(ast, {
        JSXElement(path) {
            const { openingElement } = path.node;

            // Look for `<input ... />`
            if (openingElement.name.name === 'input') {
                const isFile = openingElement.attributes.some(attr =>
                    attr.name?.name === 'type' && attr.value?.value === 'file'
                );
                const isHidden = openingElement.attributes.some(attr =>
                    attr.name?.name === 'className' && attr.value?.value?.includes('hidden')
                );

                if (isFile && isHidden) {
                    // Change className="hidden" to className="absolute inset-0 w-full h-full opacity-0 cursor-pointer z-50"
                    openingElement.attributes.forEach(attr => {
                        if (attr.name?.name === 'className') {
                            attr.value.value = "absolute inset-0 w-full h-full opacity-0 cursor-pointer z-50";
                        }
                    });

                    // Also, we need to make sure the parent container has `relative` to contain the absolute positioned input
                    // Let's go up the AST to the closest JSXElement with a className and prepend 'relative overflow-hidden '
                    let parentPath = path.parentPath;
                    while (parentPath && parentPath.node.type !== 'JSXElement') {
                        parentPath = parentPath.parentPath;
                    }
                    if (parentPath && parentPath.node.type === 'JSXElement') {
                        const parentOpeningElement = parentPath.node.openingElement;
                        const classAttr = parentOpeningElement.attributes.find(attr => attr.name?.name === 'className');
                        if (classAttr && classAttr.value && classAttr.value.type === 'StringLiteral') {
                            if (!classAttr.value.value.includes('relative')) {
                                classAttr.value.value = 'relative overflow-hidden ' + classAttr.value.value;
                            }
                        }
                    }

                    modified = true;
                }
            }
        }
    });

    if (modified) {
        const output = generate(ast, {}, code);
        fs.writeFileSync(file, output.code, 'utf8');
        updatedCount++;
        console.log(`Updated ${file}`);
    }
}

console.log(`Total files updated: ${updatedCount}`);
